piksi_tools/console/settings_list.py

```diff
@@ -1,44 +1,84 @@
+#!/usr/bin/env python
+# Copyright (C) 2011-2014 Swift Navigation Inc.
+# Contact: Fergus Noble <fergus@swift-nav.com>
+#
+# This source is subject to the license found in the file 'LICENSE' which must
+# be be distributed together with this source. All other rights reserved.
+#
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+
 import yaml
 import pprint
+import sys
+import os
+
+def determine_path ():
+    """Borrowed from wxglade.py"""
+    try:
+        root = __file__
+        if os.path.islink (root):
+            root = os.path.realpath (root)
+        return os.path.dirname (os.path.abspath (root))
+    except:
+        print "There is no __file__ variable. Please contact the author."
 
 class SettingsList():
-  list_of_dicts = list();
+  list_of_dicts = {}
 
   def __str__(self):
     pp = pprint.PrettyPrinter(indent=2)
     pp.pprint(self.list_of_dicts)
 
   def get_dict(self, group, name):
-    for element in self.list_of_dicts:
-        if element['name'] == name and element['group'] == group:
-          return element
+    # confirm our class actually has list_of_dicts
+    if self.list_of_dicts :
```

Comment:
You shouldn't need this since here - since you're walking the list below, its a spurious check.